src/docs-developer-guide.rst

```diff
@@ -122,3 +122,188 @@ For example:
+  :return: The error list consisting of error tuples: :py:const:`[(start, end, err, msg, replacement)]`.
+  :rtype: list
+
+
+.. py:function:: consistency_check(text, word_pairs, err, msg, offset=0)
+
+   To check for consistency for the given word pairs.
+   The parameters :py:data:`offset`
+   is not needed for style guide testing.
+
+   :param str text: Text to be checked
+   :param list word_pairs: Word pairs to be checked for consistency
+   :param str err: Name of the test
+   :param str msg: Error or warning message
+   :return: The error list consisting of error tuples: :py:const:`[(start, end, err, msg, replacement)]`.
+   :rtype: list
+
+.. note::
+
+  The checker functions are used by the
+  inbuilt proselint function :py:func:`lint`
+  to generate an error list of different format.
+  The returned list finally is: :py:const:`[(check, message, line, column, start, end, end - start, "warning", replacements)]`
+
+.. seealso::
+
+  To get a clear understanding,
+  refer the code for the modules `here <https://github.com/amperser/proselint/blob/master/proselint/tools.py>`_.
+
+.. rubric:: Example Usage
+
+.. code-block:: python
+
+  @memoize
+  def example(text):
+      """Example check."""
+      err = "style-guide.example"
+      msg = "A demonstration for writing checks."
+      regex = "[\.\?!](example)"
+
+      return existence_check(text, [regex], err, msg, ignore_case=False, 
+                         require_padding=False)
+
+When you define code-blocks which
+use inbuilt proselint testing,
+specify the class **style-checks**.
+
+.. code-block:: rst
+
+  .. code-block:: python
+    :class: style-checks
+
+The generated file after parsing code
+for style checks is :file:`style-checks.py`.
+
+
+If the test is too large
+to be defined in the file :file:`docs-style-guide.rst`,
+you can use a snippet from the test
+(as :ref:`here <american-spelling>`).
+The code-blocks for such snippets
+should specify the class **proselint-extra-checks**.
+Define the complete test
+in the file :file:`/style-guide/proselint-extra-checks.py`.
+
+.. _independent-checks:
+
+Independent checks
+~~~~~~~~~~~~~~~~~~~
+
+Apart from the checks, which are to be run through proselint,
+you can add extra checks to be run independently.
+The code-blocks for extra checks
+should specify the class **extra-checks**.
+The generated file after parsing code
+for extra checks is :file:`extra-checks.py`.
+
+.. note::
+
+  Inbuilt proselint function :py:func:`line_and_column` is used with extra checks to obtain the row and column of the matched text.
+
+  .. py:function:: line_and_column(text, start)
+
+    To find the line number and column of a position in a string.
+
+    :param str text: Text to be searched for
+    :param int start: Starting position of matched pattern
+    :return: Tuple containing row and column number
+    :rtype: tuple
+
+.. _classify-checks:
+
+Error vs warning
+~~~~~~~~~~~~~~~~~~
+
+You can classify the result of a check
```

Comment:
Should specify that:
 - warnings are intended to provide guidance to authors
 - errors enforce "hard" rules, and raising an error will stop the build.